salt/state.py

```diff
@@ -1775,16 +1775,36 @@ def check_requisite(self, low, running, chunks, pre=False):
 
         return status, reqs
 
-    def event(self, chunk_ret, length):
+    def event(self, chunk_ret, length, fire_event_flag=False):
         '''
         Fire an event on the master bus
+
+        If `fire_event_flag` is set to True an event will be sent with the
+        chunk name in the tag and the chunk result in the event data.
+
+        If `fire_event_flag` is set to a string such as `mystate/is/finished`,
+        an event will be sent with the string added to the tag and the chunk
+        result in the event data.
+
+        If the `state_events` is set to True in the config, then after the
+        chunk is evaluated an event will be set up to the master with the
+        results.
         '''
-        if not self.opts.get('local') and self.opts.get('state_events', True) and self.opts.get('master_uri'):
-            ret = {'ret': chunk_ret,
-                   'len': length}
-            tag = salt.utils.event.tagify(
-                    [self.jid, 'prog', self.opts['id'], str(chunk_ret['__run_num__'])], 'job'
-                    )
+        if not self.opts.get('local') and (self.opts.get('state_events', True) or fire_event_flag) and self.opts.get('master_uri'):
```

Comment:
These are difficult to read in Python (at least for me). An inline comment would help.